sdk/confidentialledger/Azure.Security.CodeTransparency/src/CodeTransparencyClient.cs

```diff
@@ -70,6 +75,60 @@ public CodeTransparencyClient(Uri endpoint, CodeTransparencyClientOptions option
         {
         }
 
+        private static List<(string IssuerHost, byte[] ReceiptBytes)> GetReceiptsFromTransparentStatementStatic(byte[] transparentStatementCoseSign1Bytes)
+        {
+            CoseSign1Message coseSign1Message = CoseMessage.DecodeSign1(transparentStatementCoseSign1Bytes);
+
+            if (!coseSign1Message.UnprotectedHeaders.
+            TryGetValue(new CoseHeaderLabel(CcfReceipt.CoseHeaderEmbeddedReceipts),
+            out CoseHeaderValue embeddedReceipts))
+            {
+                throw new InvalidOperationException("embeddedReceipts not found");
+            }
+
+            CborReader cborReader = new CborReader(embeddedReceipts.EncodedValue);
+            cborReader.ReadStartArray();
+
+            var receiptList = new List<(string, byte[])>();
+            while (cborReader.PeekState() != CborReaderState.EndArray)
+            {
+                var receipt = cborReader.ReadByteString();
+                // the receipt could be from any other system and we can only validate the ones
+                // that originate in our service instances and have an issuer we can parse
+                // identify receipts with unknown issuers with appended index to avoid clashes
+                // verification logic has a separate path to handle unknown issuers
+                string issuer = string.Empty;
+                try
+                {
+                    issuer = GetReceiptIssuerHostStatic(receipt);
+                }
+                catch (InvalidOperationException)
+                {
+                    issuer = UnknownIssuerPrefix + receiptList.Count;
```

Comment:
It's unlikely that someone will use "unknown-issuer5", but not completely impossible; using a type that cannot intersect with valid text strings/authorized issuers would be cleanest here.

Reply:
Changed to `__unknown-issuer::`, i.e. it would become `__unknown-issuer::5` in your example. `::` would not make it as it would be an invalid domain, `__` signals private use.